### ISSUE TITLE
[Feature Req] Direct calculation of growth factor for alternative dark energy models

### ISSUE THREAD

**Issue by steven-murray** [halomod/hmf#130]
**Is your feature request related to a problem? Please describe.**
It would be good to not have to have CAMB available to compute growth factors for dark energy models (plus it's slower for this since the calculation is to compute the transfer functions). 

**Describe the solution you'd like**
Implement DE.

**Comment by seVenVo1d** [comment]
Dear Steven Murray,

I am a M.Sc. Physics student in Istanbul Technical University. For my masters studies, I have tried to implement different DE parametrizations via astropy modules. In some cases I am getting errors that claims I cannot implement any other DE model then LCDM and wCDM. Is it really the case ?

**Comment by steven-murray** [comment]
Yeah, this is the case. I'd be willing to accept PRs that expand the capabilities though!

**Comment by seVenVo1d** [comment]
> Yeah, this is the case. I'd be willing to accept PRs that expand the capabilities though!

I would like to help but I am not sure where to start...But I'll try to look in my free tlmes.

I also wonder, is it possible to obtain the DE information from CAMB and later on add it to `hmf` so that I can produce plots ?

**Comment by steven-murray** [comment]
I think the main thing you need to do is make sure your transfer function and the growth factor calculation are correct. You can do that however you like, either using in-built classes in `hmf` or adding your own. The `FromArray` and `FromFile` options might be an easy patch for now, if you can get the info from camb.

**Comment by seVenVo1d** [comment]
I see your point. Thanks for the help. I will try to perform both approaches, but first, I am going to try to add the equations in the `hmf`.

**Comment by steven-murray** [comment]
Scale-independent growth factors for any dark energy model are now implemented via the full ODE in #270
